SeaPublicWebsite/Services/RecommendationService.cs

```diff
@@ -300,24 +300,61 @@ private static BrePropertyType GetBrePropertyType(PropertyType propertyType)
             }
         }
 
-        private static string GetBreConstructionDate(int? yearBuilt)
+        private static string GetBreConstructionDate(YearBuilt? yearBuilt, WallConstruction? wallConstruction, CavityWallsInsulated? cavityWallsInsulated,  HomeAge? epcConstructionAgeBand)
         {
             return yearBuilt switch
             {
-                <= 1899 => "A",
-                <= 1929 => "B",
-                <= 1949 => "C",
-                <= 1966 => "D",
-                <= 1975 => "E",
-                <= 1982 => "F",
-                <= 1990 => "G",
-                <= 1995 => "H",
-                <= 2002 => "I",
-                <= 2006 => "J",
-                <= 2011 => "K",
-                >= 2012 => "L",
-                //peer-reviewed assumption:
-                _ => "D"
+                YearBuilt.Pre1930 => "B",
+                YearBuilt.From1930To1966 => "D",
+                YearBuilt.From1967To1982 => "F",
+                YearBuilt.From1983To1995 => "H",
+                YearBuilt.From1996To2011 => "K",
+                YearBuilt.From2012ToPresent => "L",
+                //peer-reviewed assumptions:
+                _ => wallConstruction switch
+                {
+                    WallConstruction.DoNotKnow => epcConstructionAgeBand switch
+                    {
+                        HomeAge.Pre1900 => "A",
+                        HomeAge.From1900To1929 => "B",
+                        HomeAge.From1930To1949 => "C",
+                        HomeAge.From1950To1966 => "D",
+                        HomeAge.From1967To1975 => "E",
+                        HomeAge.From1976To1982 => "F",
+                        HomeAge.From1983To1990 => "G",
+                        HomeAge.From1991To1995 => "H",
+                        HomeAge.From1996To2002 => "I",
+                        HomeAge.From2003To2006 => "J",
+                        HomeAge.From2007ToPresent => "L",
+                        _ => "D",
+                    },
+                    WallConstruction.Solid => "B",
+                    WallConstruction.Cavity => cavityWallsInsulated switch
+                    {
+                        CavityWallsInsulated.DoNotKnow => epcConstructionAgeBand switch
+                        {
+                            HomeAge.Pre1900 => "A",
+                            HomeAge.From1900To1929 => "B",
+                            HomeAge.From1930To1949 => "C",
+                            HomeAge.From1950To1966 => "D",
+                            HomeAge.From1967To1975 => "E",
+                            HomeAge.From1976To1982 => "F",
+                            HomeAge.From1983To1990 => "G",
+                            HomeAge.From1991To1995 => "H",
+                            HomeAge.From1996To2002 => "I",
+                            HomeAge.From2003To2006 => "J",
+                            HomeAge.From2007ToPresent => "L",
```

Comment:
no K, there is a 2007-2012 band that isn't included in the ticket, should we add this?